client/hijack.go

```diff
@@ -113,7 +115,12 @@ func (cli *Client) setupHijackConn(ctx context.Context, req *http.Request, proto
 		br.Reset(nil)
 	}
 
-	return c, nil
+	if versions.GreaterThanOrEqualTo(cli.ClientVersion(), "1.42") {
+		mediaType := resp.Header.Get("Content-Type")
+		b := mediaType == types.MediaTypeMultiplexedStream
+		return c, &b, nil
+	}
```

Comment:
I don't think we need this; the essence of @cpuguy83's comment is that "older clients" == "existing clients" and/or "clients we didn't create". So clients that we cannot upgrade, so we should make sure the API server returns the expected type for those (when using API < 1.42), and for newer API versions return the _correct_ mediaType.

Reply:
That's exactly my intent here: client and server for api < 1.42 don't rely on Content-Type to guess the stream encoding. API 1.42 introduces a new media-type to help distinguish those.

Reply:
🤦 ah, right, yes. I started looking at the changes from my phone, then continue to my computer 🤦


So the `application/vnd.docker.multiplexed-stream` content-type is only returned by the daemon if it supports it, and (should) only be returned if the client sent it as `Accept` header (older versions of the API didn't know about that type)

Which means that for the "positive" case (`application/vnd.docker.multiplexed-stream` was returned) we wouldn't need the API version, but for the `application/vnd.docker.raw-stream` we do :thinking:


I guess the alternative would be to use two completely new types that are only available on daemons that support this (`application/vnd.docker.raw-stream-v2`, `application/vnd.docker.multiplexed-stream`). Clients that don't have an accept header (or only accept `application/vnd.docker.raw-stream`) would get the old type, and clients that accept `application/vnd.docker.raw-stream-v2`, `application/vnd.docker.multiplexed-stream` would get one of the new types.